ui/src/pages/domain/[domain]/service/[service]/microsegmentations.js

```diff
@@ -0,0 +1,235 @@
+`;
+
+const MicrosegmentationsContentDiv = styled.div``;
+
+const PageHeaderDiv = styled.div`
+    background: linear-gradient(to top, #f2f2f2, #fff);
+    padding: 20px 30px 0;
+`;
+
+export async function getServerSideProps(context) {
+    let api = API(context.req);
+    let reload = false;
+    let notFound = false;
+    let error = null;
+    const microsegmentationsData = await Promise.all([api.getForm(), api.getPageFeatureFlag('microsegmentation')]).catch((err) => {
+        let response = RequestUtils.errorCheckHelper(err);
+        reload = response.reload;
+        error = response.error;
+        return [{}];
+    });
+    return {
+        props: {
+            reload,
+            notFound,
+            error,
+            serviceName: context.query.service,
+            userName: context.req.session.shortId,
+            _csrf: microsegmentationsData[0],
+            domainName: context.query.domain,
+            nonce: context.req.headers.rid,
+            pageFeatureFlag: microsegmentationsData[1],
+        },
+    };
+}
+
+class ServiceMicrosegmentationsPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = API();
+        this.cache = createCache({
+            key: 'athenz',
+            nonce: this.props.nonce,
+        });
+        this.state = {
+            error: null,
+            reload: false,
+            errorMessage: '',
+            showError: false,
+        };
+    }
+
+    componentDidMount() {
+        const { domainName, userName, getDomainData, getInboundOutbound } = this.props;
+        Promise.all([
+            getDomainData(domainName, userName),
+            getInboundOutbound(domainName),
+        ]).catch((err) => {
+            let response = RequestUtils.errorCheckHelper(err);
+            this.setState({
+                error: response.error,
+                reload: response.reload,
+            });
+        });
+    }
+
+    showError(errorMessage) {
+        this.setState({
+            showError: true,
+            errorMessage: errorMessage,
+        });
+    }
+
+    render() {
+        const {
+            domainName,
+            reload,
+            serviceName,
+            serviceDetails,
+            _csrf,
+            isLoading,
+        } = this.props;
+        if (reload || this.state.reload) {
+            window.location.reload();
+            return <div />;
+        }
+        const err = this.props.error || this.state.error;
+        if (err) {
+            return <Error err={err} />;
+        }
+
+        if (this.state.showError) {
+            return (
+                <Alert
+                    isOpen={this.state.showError}
+                    title={this.state.errorMessage}
+                    onClose={() => {}}
+                    type='danger'
+                />
+            );
+        }
+
+        return isLoading.includes('getDomainData') ? (
+            <ReduxPageLoader message={'Loading domain data'} />
```

Comment:
should this just be `isLoading.length > 0 ? <ReduxPageLoader .../>` ? 

Basically should we load while waiting for response from `getInboundOutbound` ?